src/core/thread/network_diagnostic_tlvs.hpp

```diff
@@ -1348,6 +1348,24 @@ class ChildTableTlv : public NetworkDiagnosticTlv
         return *reinterpret_cast<ChildTableEntry *>(GetValue() + (aIndex * sizeof(ChildTableEntry)));
     }
 
+    /**
+     * This method reads the Child Table entry at @p aIndex.
+     *
+     * @param[in]  aIndex  The index into the Child Table list.
+     *
+     * @returns  if read successfully.
+     */
+    otError ReadEntry(ChildTableEntry &aEntry, Message &aMessage, uint16_t aOffset, uint8_t aIndex)
```

Comment:
Could existing `GetEntry()` in Line 1346 serve the same purpose?

Reply:
discussed offline, `GetEntry()` cannot serve it.